### ISSUE TITLE
codegen generate wrong signature

### ISSUE THREAD

**Issue by vtugaye** [sympy/sympy#16689]
When an argument sequence is specified in codegen, any matrix argument that is not present in the expression to be converted will get a default type `double` instead of `double*`.
For instance :

```python
import sympy as sy
from sympy.utilities.codegen import codegen

X = sy.MatrixSymbol('X',3,1)
Y = sy.MatrixSymbol('Y',3,1)
z = sy.symbols('z',integer = True)

[(cf, cs), (hf, hs)] = codegen(('testBug', X[0]+X[1]),
           language='c',argument_sequence = (X,Y,z))
print(hs)
```

generates  a function signature : `double testBug(double *X, double Y, int z);`

where `double testBug(double *X, double *Y, int z);`  is expected

But 

```python
[(cf, cs), (hf, hs)] = codegen(('testBug', X[0]+X[1]+Y[0]),
           language='c',argument_sequence = (X,Y,z))
print(hs)
```
gives the good signature.

I use sympy 1.3

**Comment by wermos** [comment]
I just tried this with the current [`master`](https://github.com/sympy/sympy/commit/e2cadc140cc969fea038240a39961a66a2f3dd6d) and it worked as expected.

```python
>>> import sympy as sy
>>> from sympy.utilities.codegen import codegen
>>> X = sy.MatrixSymbol('X',3,1)
>>> Y = sy.MatrixSymbol('Y',3,1)
>>> z = sy.symbols('z',integer = True)
>>> [(cf, cs), (hf, hs)] = codegen(('testBug', X[0]+X[1]),
...            language='c',argument_sequence = (X,Y,z))
>>> print(hs)
/******************************************************************************
 *                     Code generated with SymPy 1.13.dev                     *
 *                                                                            *
 *              See http://www.sympy.org/ for more information.               *
 *                                                                            *
 *                       This file is part of 'project'                       *
 ******************************************************************************/


#ifndef PROJECT__TESTBUG__H
#define PROJECT__TESTBUG__H

double testBug(double *X, double *Y, int z);

#endif
```

I think we can mark this issue as solved.

cc @moorepants for thoughts.

**Comment by wermos** [comment]
It appears to have been fixed in b56caac90a903fbb0713cd56119a4bbd6ca3ad8c as a part of #16792 so I think we can close this issue as already solved.

**Comment by moorepants** [comment]
We can close this, but it would also be helpful to add a short regression test for this. The code from this issue can be used.
